crates/sui-core/src/authority.rs

```diff
@@ -4549,14 +4543,25 @@ impl AuthorityState {
             .execution_lock_for_executable_transaction(&executable_tx)
             .await?;
 
+        epoch_store
+            .process_consensus_transaction_shared_object_versions(
+                self.get_cache_reader().as_ref(),
+                &[executable_tx.clone()],
+                None,
+            )
+            .await?;
```

Comment:
This is not crash-safe yet. We must not reassign this transaction twice in case of crash.
Let me think about what we should do about it and whether this is still the right PR to do.
@mystenmark If you have an opinion please feel free to comment

Reply:
I think maybe what you had before that doesn't update next_shared_object_versions table is the right thing? In that case, if the transaction is already assigned versions and then the validator crashes, because next_shared_object_versions is not updated, it will assign the same version again when the transaction is processed the second time (even though the assigned version may already be inserted into assigned_shared_object_versions table, it'll just be inserted again).